### ISSUE TITLE
osu!lazer won't run after 2019.1011.0 update

### ISSUE THREAD

**Issue by Sinamer** [ppy/osu#6496]
**Describe the crash:** It returns system error upon launch

**Screenshots or videos showing encountered issue:** 
![Screenshot_1](https://user-images.githubusercontent.com/53346335/66717059-85366280-eddd-11e9-8bd3-4925fa80b7d3.png)

**osu!lazer version:** 2019.1011.0

**Logs:** Not sure if it gets logged, here's the only one that was updated
[runtime.log](https://github.com/ppy/osu/files/3721849/runtime.log)
 

**Computer Specifications:** 
Windows 10 Pro build 16299, 64-bit
CPU: Intel(R) Celeron(R) CPU B800
GPU: Intel(R) HD Graphics 
2gb RAM

**Comment by bdach** [comment]
The OpenGL initialization header is concerning:
```
2019-10-13 02:21:43: GL Initialized
2019-10-13 02:21:43: GL Version:                 1.1.0
2019-10-13 02:21:43: GL Renderer:                GDI Generic
2019-10-13 02:21:43: GL Shader Language version:
2019-10-13 02:21:43: GL Vendor:                  Microsoft Corporation
2019-10-13 02:21:43: GL Extensions:              GL_WIN_swap_hint GL_EXT_bgra GL_EXT_paletted_texture
```
A generic GDI renderer should *not* be used by the game with your GPU. Are you sure Windows hasn't installed generic drivers for your card? Do you mind trying to update them?

Although it *is* weird that this would surface after upgrading - does downgrading again help? If it does, mind attaching log output from the previous version?

**Comment by Sinamer** [comment]
Downgrading to 2019.930.0 again helps, here are the logs:
[performance.log](https://github.com/ppy/osu/files/3721880/performance.log)
[runtime.log](https://github.com/ppy/osu/files/3721881/runtime.log)
[updater.log](https://github.com/ppy/osu/files/3721882/updater.log)
[network.log](https://github.com/ppy/osu/files/3721884/network.log)

**Comment by bdach** [comment]
Looks good on the old version, weirdly enough the reported *system version* changes too...?
```
2019-10-13 02:58:17: GL Initialized
2019-10-13 02:58:17: GL Version:                 3.1.0 - Build 9.17.10.4459
2019-10-13 02:58:17: GL Renderer:                Intel(R) HD Graphics
2019-10-13 02:58:17: GL Shader Language version: 1.40 - Intel Build 9.17.10.4459
2019-10-13 02:58:17: GL Vendor:                  Intel
```
Do you mind checking the Windows Event Viewer for any potentially related errors around the time of launch of the new version? Sorry for the hassle, but this issue seems rare (as it hasn't been reported yet) and particularly bizarre.

**Comment by Sinamer** [comment]
There is an error, it says: 

Application: osu!.exe
CoreCLR Version: 4.700.19.46205
.NET Core Version: 
Description: The process was terminated due to an internal error in the .NET Runtime at IP 00007FFDD3E64523 (00007FFDD3CB0000) with exit code c0000005.

Tell me if I understood it wrong

**Comment by bdach** [comment]
Thanks for looking - unfortunately this is virtually the same error as the dialog shows, and is similarly unhelpfully non-specific, at least to me - all it says is that there was an access violation (`0xc0000005`) at the given instruction pointer.

**Comment by enoslayd** [comment]
i also suffered from this issue now when updated to 1011.0 but it doesnt have any dialog box
[performance.log](https://github.com/ppy/osu/files/3723259/performance.log)
[runtime.log](https://github.com/ppy/osu/files/3723260/runtime.log)
[Uploading updater.log…]()
[Uploading database.log…]()
[Uploading network.log…]()

**Comment by enoslayd** [comment]
i also suffered from this issue now when updated to 1011.0 but it doesnt have any dialog box
[performance.log](https://github.com/ppy/osu/files/3723259/performance.log)
[runtime.log](https://github.com/ppy/osu/files/3723260/runtime.log)
[updater.log](https://github.com/ppy/osu/files/3723261/updater.log)
[database.log](https://github.com/ppy/osu/files/3723262/database.log)
[network.log](https://github.com/ppy/osu/files/3723263/network.log)

**Comment by bdach** [comment]
@enoslayd Please do not post double comments. Do you mind giving your PC specifications (in particular the GPU make and model)?

**Comment by enoslayd** [comment]
sorry. i didnt know what happenned though

**Comment by enoslayd** [comment]
intel core i3 540m 3.07 ghz
no gpu (intel hd graphics only)
2GB ram
256GB SSD

i will try this update on my laptop later

**Comment by peppy** [comment]
This has got to be related to the net core 3.0 update, but very weird that GL is detecting different underlying hardware..

**Comment by smoogipoo** [comment]
As for the version being different, that's expected with .NET Core 3.0, see: https://github.com/dotnet/coreclr/issues/22844 /  https://devblogs.microsoft.com/dotnet/announcing-net-core-3-0/ . This may require an osuTK fix.

**Comment by enoslayd** [comment]
in my laptop, there is this window "osu framework running "osu"" and it still crashing

Specs:
Intel pentium P6200 2.13 GHz dual core
4 GB ram
No gpu (intel hd graphics only)
250 GB SSD

[runtime.log](https://github.com/ppy/osu/files/3727229/runtime.log)

**Comment by Joppe27** [comment]
I have the same issue, osu! doesn't launch since 2019.1011.0 update. System specs so you guys can get some more info: 
- Processor:	Intel(R) Core(TM) i5 CPU       M 520  @ 2.40GHz
- GPU: Intel HD Graphics (Core i5); Intel64 Family 6 Model 37 Stepping 5
- RAM: 4GB
- 250 GB SSD
Hope this helps you out in finding out what causes this error.
[http://s000.tinyupload.com/index.php?file_id=05175951822003466966](logs)

**Comment by MeguMario** [comment]
For those who really want to try out 2019.1011, 2019.1021.0 and 2019.1029.0 on the affected machines, there is a way that will involve system files replacing, so do this if you really want to try it

Credit: https://github.com/libretro/RetroArch/issues/4580#issuecomment-305754318

Files
[Intel_OpenGL_Win10_9.17.10.4459_Fix_x64.zip](https://github.com/ppy/osu/files/3790592/Intel_OpenGL_Win10_9.17.10.4459_Fix_x64.zip)

2019.1029.0
![image](https://user-images.githubusercontent.com/44724359/67887092-3c243380-fb7d-11e9-8d3a-56a2eaedcfb2.png)
2019.1021.0
![image](https://user-images.githubusercontent.com/44724359/67887681-5579af80-fb7e-11e9-905f-d4322efd64d3.png)

Seems like the driver is missing both ig7icd64.dll and ig7icd32.dll

**Comment by bdach** [comment]
@MeguMario Thanks for linking that actually, it's very helpful! Looking through the threads I stumbled upon [this comment in the LWJGL repository](https://github.com/LWJGL/lwjgl/issues/119#issuecomment-177218067), which seems to indicate this is Intel device driver breakage - it seems to die if the major Windows version is not 5 or 6.

I had a look at the Windows binaries pre- and post-2019.930.0 inside a hex editor, and found the manifest before and after the .NET Core 3.0 bump.

* 2019.930.0 has no `<compatibility>` element in the manifest.
* Newer releases have a `<compatibility>` element, *and* declare compatibility with Windows 10.

Since I can't reproduce I can't confirm, but if we found a way to binary patch that element out of the releases, then this could maybe go away, if what the comment claims is correct (ie. the manifest is used to get the version)? Don't get me wrong, it's a *real* hack for some prime Intel breakage.

It could also be not enough, and the root cause could be .NET Core changing the returned Windows version altogether, in which case short of reversing the actual drivers and binary patching the check out of it (I... don't see that happening) we might be screwed.

In summary, Windows version tests are evil and don't ever do them.

**Comment by MeguMario** [comment]
@bdach as far as i know, this OpenGL bug only affected Sandy Bridge iGPU and older (oldest is GMA X3100, which have at least OpenGL 2.0, though you have to use driver Intel supplied, since Microsoft only provide generic driver with no OpenGL capabilities and older), but that is for java side, the bug isn't affect Ivy Bridge iGPU, at least for the current driver.

Though I believed that it was the driver faults, what you have mention is likely to be fixed via a patch, though it also can be unlikely

I should also mention that GMA X3100 driver (till X4500/Mobile Intel 4) also break OpenGL on Windows 8.1 as well, but that is also for java side

**Comment by MeguMario** [comment]
> @MeguMario Thanks for linking that actually, it's very helpful! Looking through the threads I stumbled upon [this comment in the LWJGL repository](https://github.com/LWJGL/lwjgl/issues/119#issuecomment-177218067), which seems to indicate this is Intel device driver breakage - it seems to die if the major Windows version is not 5 or 6.
> 
> I had a look at the Windows binaries pre- and post-2019.930.0 inside a hex editor, and found the manifest before and after the .NET Core 3.0 bump.
> 
> * 2019.930.0 has no `<compatibility>` element in the manifest.
> * Newer releases have a `<compatibility>` element, _and_ declare compatibility with Windows 10.
> 
> Since I can't reproduce I can't confirm, but if we found a way to binary patch that element out of the releases, then this could maybe go away, if what the comment claims is correct (ie. the manifest is used to get the version)? Don't get me wrong, it's a _real_ hack for some prime Intel breakage.
> 
> It could also be not enough, and the root cause could be .NET Core changing the returned Windows version altogether, in which case short of reversing the actual drivers and binary patching the check out of it (I... don't see that happening) we might be screwed.
> 
> In summary, Windows version tests are evil and don't ever do them.

I should also mention that some driver even broke on Windows 8/8.1 (mostly GMA x4500 and lower)

**Comment by huoyaoyuan** [comment]
Have you tried to run it in compatibility mode?

<img width="674" alt="image" src="https://user-images.githubusercontent.com/5644458/67928949-2bb09f00-fbf7-11e9-8706-13ffcd795e53.png">

Windows will report your selected OS version in compatibility mode.

**Comment by Sinamer** [comment]
I still get a crash with Windows 7 compatibility mode
No logs, two errors in Event Viewer:
https://drive.google.com/open?id=1HFe6KuqKDsLrLxElm80EAjg92UGSOOH0

**Comment by MeguMario** [comment]
> ave you trie

sadly it seems to be the driver, compatbility mode won't get you anywhere

**Comment by pal1000** [comment]
Here is a video tutorial that describe all possible fixes to this problem with full demonstration of compatibility shim method which I believe is best of all
[![Play](https://pal1000.github.io/shared/tutorials/video.png)](https://pal1000.github.io/shared/tutorials/osulaser-fix-demo.mp4)

**Comment by bdach** [comment]
@pal1000 Thanks for the video and for confirming that removing the manifest entries does indeed fix it! It's a band-aid workaround, but in my opinion it's the best way for us to provide out-of-the-box compatibility with older Intel cards, unless it turns out the compatibility entries help with something else. I'm pretty sure we should be able to modify the manifest at source-level, which would end up propagating to the released binaries.

(The irony of added compatibility entries making the game less compatible with hardware due to broken Intel drivers that they won't fix is not lost on me.)

**Comment by bdach** [comment]
Could anyone who has this problem test out #7088 and confirm for sure that it fixes the problem? Would be much appreciated, as I don't have the appropriate hardware on hand.

**Comment by doughn0** [comment]
I can confirm that it works for me, I have an Intel HD 3000.

Old log:
[runtime.log](https://github.com/ppy/osu/files/3935520/runtime.log)

Log from that pull:
[runtime.log](https://github.com/ppy/osu/files/3935522/runtime.log)

**Comment by bdach** [comment]
@doughn0 Great to hear, thanks so much for checking!
